From 6cb24cea619a9041e1c3fad46f62e97cf561e522 Mon Sep 17 00:00:00 2001
From: Vincas Dargis <vindrg@gmail.com>
Date: Tue, 7 Aug 2018 12:23:48 +0300
Subject: deb#905442 apparmor: fix saving files in enforced mode

Add rule to allow creating temporary files like `lu6924jgpjkq.tmp`,
because without it files cannot be saved.

Change-Id: I640897df5d325e2d01da16840034abbd095fb276
Reviewed-on: https://gerrit.libreoffice.org/58672
Tested-by: Jenkins
Reviewed-by: Miklos Vajna <vmiklos@collabora.co.uk>
(cherry picked from commit 8ca83479ab1f921a14a366432a513a904af0dd20)
Reviewed-on: https://gerrit.libreoffice.org/58677
Reviewed-by: Rene Engelhard <rene@debian.org>
Tested-by: Rene Engelhard <rene@debian.org>
---
 sysui/desktop/apparmor/program.soffice.bin | 1 +
 1 file changed, 1 insertion(+)

diff --git a/sysui/desktop/apparmor/program.soffice.bin b/sysui/desktop/apparmor/program.soffice.bin
index 2fc7fd6..a680260 100644
--- a/sysui/desktop/apparmor/program.soffice.bin
+++ b/sysui/desktop/apparmor/program.soffice.bin
@@ -97,6 +97,7 @@ profile libreoffice-soffice INSTDIR-program/soffice.bin {
   owner @{libo_user_dirs}/**/           rw,  #allow creating directories that we own
   owner @{libo_user_dirs}/**~lock.*     rw,  #lock file support
   owner @{libo_user_dirs}/**.@{libreoffice_ext} rwk,  #Open files rw with the right exts
+  owner @{libo_user_dirs}/{,**/}lu??????????{,?}.tmp rwk, #Temporary file used when saving
 
   # Settings
   /etc/libreoffice/                     r,
-- 
cgit v1.1

